pennylane/pauli/conversion.py

```diff
@@ -14,13 +14,133 @@
 """
 Utility functions to convert between ``~.PauliSentence`` and other PennyLane operators.
 """
+from typing import Union
+from operator import matmul
+from itertools import product
 from functools import reduce, singledispatch
 
+import numpy as np
+
 from pennylane.operation import Tensor
 from pennylane.ops import Hamiltonian, Identity, PauliX, PauliY, PauliZ, Prod, SProd, Sum
 
-from .pauli_arithmetic import PauliWord, PauliSentence, X, Y, Z
 from .utils import is_pauli_word
+from .pauli_arithmetic import PauliWord, PauliSentence, I, X, Y, Z, mat_map, op_map
+
+
+def pauli_decompose(
+    H, hide_identity=False, wire_order=None, pauli=False
+) -> Union[Hamiltonian, PauliSentence]:
+    r"""Decomposes a Hermitian matrix into a linear combination of Pauli operators.
+
+    Args:
+        H (array[complex]): a Hermitian matrix of dimension :math:`2^n\times 2^n`.
+        hide_identity (bool): does not include the Identity observable within
+            the tensor products of the decomposition if ``True``.
+        wire_order (list[Union[int, str]]): the ordered list of wires with respect
+            to which the operator is represented as a matrix.
+        pauli (bool): return a PauliSentence instance if ``True``.
+
+    Returns:
+        Union[~.Hamiltonian, ~.PauliSentence]: the matrix decomposed as a linear combination
+            of Pauli operators, either as a ``~.Hamiltonian`` or ``~.PauliSentence`` instance.
+
+    **Example:**
+
+    We can use this function to compute the Pauli operator decomposition of an arbitrary Hermitian
+    matrix:
+
+    >>> A = np.array(
```

Comment:
Again pure curiosity: Is this decomposition unique? Or is this _a_ decomposition?

Reply:
I think this decomposition is unique, since a vector is uniquely represented in a given basis and the n tensor product of pauli operators form a [complete basis](https://quantumcomputing.stackexchange.com/questions/2703/is-the-pauli-group-for-n-qubits-a-basis-for-mathbbc2n-times-2n) for complex matrices of size 2^n x 2^n.

Reply:
Right! Thanks! :)